src/cpp/session/prefs/UserPrefsLayer.cpp

```diff
@@ -109,6 +109,19 @@ Error UserPrefsLayer::writePrefs(const core::json::Object &prefs)
       lastSync_ = prefsFile_.getLastWriteTime();
    }
 
+   // Modify the error to be more descriptive
+   if (isFileNotFoundError(error) && prefsFile_.exists())
+   {
+      error = Error(
```

Comment:
I wonder if we should instead amend the existing error's properties here, rather than creating a whole new error object? (Since in theory some properties of interest on the original error could be lost)

Reply:
We could even add a helper method to make this a sort of "official" way of tagging this kind of information on an error (since there's a lot of places in the code base where users and support would benefit from extra tagged information). Or do we have an existing one that is suitable here?

Reply:
The dialog that the user is presented with only displays the message of the error. All the properties are left out. The only way to set the message of  an error is at construction time. However, I do think it would be reasonable to set the original error as the cause, or at least copy over all the properties.

Reply:
What do you think about setting or amending the message on the original error object?

Sorry for being nitpicky; I'm just trying to think if there's a sort of "best practice" that we could encode here and then reuse in other locations where we want to surface extra information for the user.

Reply:
I'm not sure. When I modified the `Error` class I continued to follow the existing pattern - i.e. that the message can't be updated after the error has been created. We use to take the message solely from the boost error code object. I don't know the reasoning behind the original desicion so I can't say whether it should be changed or not. @jmcphers Do you have any opinion on this?